pyatv/protocols/mrp/__init__.py

```diff
@@ -823,6 +837,41 @@ async def volume_down(self) -> None:
             )
             await asyncio.wait_for(self._volume_event.wait(), timeout=5.0)
 
+    async def _update_output_devices(self, message: protobuf.ProtocolMessage) -> None:
+        inner = cast(protobuf.DeviceInfoMessage, message.inner())
+        devices = []
+        if inner.isGroupLeader and not inner.isProxyGroupPlayer:
+            devices.append(OutputDevice(inner.name, inner.uniqueIdentifier))
+        for device in list(inner.groupedDevices):
+            devices.append(OutputDevice(device.name, device.deviceUID))
+        self._output_devices = devices
+        self._output_devices_event.set()
+        self._output_devices_event.clear()
+        self.state_dispatcher.dispatch(UpdatedState.OutputDevices, devices)
+
+    @property
+    def output_devices(self) -> List[OutputDevice]:
+        """Return current list of output device IDs."""
+        return self._output_devices
+
+    async def add_output_devices(self, *devices: List[str]) -> None:
+        """Add output devices."""
+        await self.protocol.send(messages.add_output_devices(*devices))
+        with suppress(asyncio.TimeoutError):
```

Comment:
Are these errors frequent? Generally I prefer if errors bobble up so users are aware of an issues rather than hiding them.

Reply:
This will happen any time we don't get a DEVICE_INFO_UPDATE_MESSAGE after sending the change request, basically any time the request doesn't cause an actual change. It would be harder to always keep track of when a change will or will not happen.

I opted to suppress it since users seem to find it confusing (see e.g.: #1838; it's not the command that timed out, but the response).

Reply:
Ok, that is good enough then!